agent/api/task.go

```diff
@@ -118,15 +119,19 @@ type Task struct {
 	// used to look up the credentials for task in the credentials manager
 	credentialsID     string
 	credentialsIDLock sync.RWMutex
+
+	// memoryCPULimits to determine if task supports CPU, memory limits
+	memoryCPULimits     bool
```

Comment:
We want to avoid passing in config from `engine` while updating the containers `CgroupParent`. 
Now during `PostUnmarshalTask`, this attribute would be updated and `DockerHostConfig` would use this. 

All hacks are localized